### PR TITLE
Add parent fields for categories returned by the proxy mode

### PR DESCRIPTION
**Pull request by mrodm**
This PR adds the missing fields related to subcategories in the categories entrypoint with proxy mode.

Currently, the output was missing `parentId`and `parentTitle`. Example:
```json
  {
    "id": "productivity",
    "title": "Productivity",
    "count": 1
  },
  {
    "id": "productivity_security",
    "title": "Productivity",
    "count": 9
  },
```

Adding those fields to the struct, the same request:
```json
  {
    "id": "productivity",
    "title": "Productivity",
    "count": 1
  },
  {
    "id": "productivity_security",
    "title": "Productivity",
    "count": 9,
    "parent_id": "security",
    "parent_title": "Security"
  },
```

## How to test this change

Test with the current production version:
- In one terminal run
  ```bash
  docker run --rm -it -p 8888:8080 -e EPR_FEATURE_PROXY_MODE=true -e EPR_PROXY_TO=https://epr.elastic.co  docker.elastic.co/package-registry/package-registry:v1.19.0
  ```
- From another terminal, test categories entrypoint for that EPR container:
  ```bash
  curl -s http://localhost:8888/categories
  ```

Test with the changes proposed in this PR:
- Build a new docker image:
  ```bash
  docker build --no-cache -t docker.elastic.co/package-registry/package-registry:v1.19.1 -f Dockerfile .
  ```
- In one terminal run
  ```bash
  docker run --rm -it -p 8888:8080 -e EPR_FEATURE_PROXY_MODE=true -e EPR_PROXY_TO=https://epr.elastic.co  docker.elastic.co/package-registry/package-registry:v1.19.1
  ```
- From another terminal, test categories entrypoint for that EPR container:
  ```
  curl -s http://localhost:8888/categories
  ```

### CODE REVIEW

File: categories.go [hunk 1]
```diff
@@ -72,9 +72,11 @@ func categoriesHandlerWithProxyMode(logger *zap.Logger, indexer Indexer, proxyMo
...
 				if _, ok := categories[category.Id]; !ok {
 					categories[category.Id] = &packages.Category{
-						Id:    category.Id,
-						Title: category.Title,
-						Count: category.Count,
+						Id:          category.Id,
+						Title:       category.Title,
+						Count:       category.Count,
+						ParentId:    category.ParentId,
+						ParentTitle: category.ParentTitle,
```
Comment (jsoriano) [inline categories.go:79]: Can we add some test for this?

**Review by jsoriano — COMMENTED** [review]
Looks good

**Review by jsoriano — APPROVED** [review]

File: categories.go (hunk 1 above)
Comment (mrodm) [reply categories.go:79]: It's not so easy to setup the scenario to test endpoints with the proxyMode.
Created issue for this #1007

Comment (mrodm) [reply categories.go:79]: Added a new file for test `proxy_mode_test.go` for these tests with proxy mode enabled.

**Review by jsoriano — APPROVED** [review]
:+1:

**Pull request merged**